templates/index.html

```diff
@@ -142,26 +127,48 @@ <h4>These states love music:</h4>
 <script src="https://d3js.org/d3-scale-chromatic.v1.min.js"></script>
 <script src="https://d3js.org/topojson.v1.min.js"></script><!-- topojson library -->
 <script src="{{url_for('static',filename = 'js/uStates.js')}}"></script><!-- creates uStates. -->
+<script src="{{url_for('static',filename = 'js/statistics.js')}}"></script><!-- creates statistics in side bar. -->
 <script>
-	uStates.draw("top")
+	//tooltip function for top genres
+	function tooltip_topgenre(name, abbr, value, genre) {
+		/* function to create html content string in tooltip div. */
+		return "<h4>" + name + ", " + abbr + "</h4>" +
+			"<table><tr><td> Top Genre:</td><td>" + genre + "</td></tr>" +
+			"</table>" +
+			"<small>(click to zoom)</small>";
+	}
+
+	//tooltip function for all genres
+	function tooltip_genre(name, abbr, value, genre) {
+		/* function to create html content string in tooltip div. */
+		return "<h4>" + name + ", " + abbr + "</h4>" +
+			"<table><tr><td>" + genre + "</td><td>" + value * 100 + "%</td></tr>" +
+			"</table>" +
+			"<small>(click to zoom)</small>";
+	}
+
+	uStates.draw("topgenre", tooltip_topgenre)
```

Comment:
[RC]
Why are you passing the tooltip function as a parameter to draw instead of defining it inside uStates.draw()?

Reply:
Mostly because I haven't worked out how I'm going to handle the tooltip with the pie chart on the main page. But you're right, I should define the tooltip for all the genres inside uStates.draw(). I assume the pie chart will be defined in a separate js file and we will need to call it in a separate function? Or is there a better way?